executor/src/witgen/identity_processor.rs

```diff
@@ -82,6 +83,17 @@ impl<'a, 'b, T: FieldElement> IdentityProcessor<'a, 'b, T> {
             .map(|e| rows.evaluate(e))
             .collect::<Vec<_>>();
 
+        // Fail if the LHS has an error.
+        let (left, errors): (Vec<_>, Vec<_>) = left.into_iter().partition_map(|x| match x {
```

Comment:
Can you use ` .partition(Result::is_ok)`?

Reply:
Yes, but this doesn't unpack the `Result`, so I'd have to do another `let left = left.into_iter().map(Result::unwrap).collect::<Vec<_>>();` (and also unpack the errors). Think it's clearer this way.